language/src/regent/optimize_index_launches.t

```diff
@@ -247,10 +263,14 @@ local function analyze_noninterference_previous(
         -- Index non-interference is handled at the type checker level
         -- and is captured in the constraints.
     then
-      return false, i
+      table.insert(args_interfering, i)
```

Comment:
If you make `args_interfering` into a `terralib.newlist()` then you can do `args_interfering:insert(i)` here instead of needing to explicitly invoke `table.insert`.

Reply:
I'm attaching the potential changes as a patch file. The two issues are that i) I still use `table.insert` in other places, and ii) I couldn't find a nicer way than using `:exists(...)` to check if a Terra list is empty.
[patch.txt](https://github.com/StanfordLegion/legion/files/6856922/patch.txt)

Reply:
A Terra list is still a Lua table with keys starting at 1. So the other code here doesn't need to change, you can still use `ipairs` or `pairs` to iterate it, and so on. Of course if you prefer the functional style you're welcome to use `:map` if you want to, but it's not necessary.

Reply:
Ok, thanks, that makes sense. I just pushed this. I stuck to using `pairs` for consistency.